Feature/refactor change signature calculation 2

kindlich opened the pull request:
Successor to #104

kindlich committed:
Properly compile Enum Constructor

kindlich committed:
Write default enum ctor if not present

kindlich committed:
Add Test for enum contextMember

kindlich committed:
Added ToDo to Enum_VALUES Bultin what needs to change in JMBC

kindlich committed:
Add isEnum() to DefinitionSymbol

kindlich committed:
WIP: Add signature to JavaCompilingMethod

kindlich committed:
WIP: Add new ctor to JavaWriter that no longer accepts a signature

Soon all other ctors that accepts signatures will be removed/changed to point to this one

kindlich committed:
Remove Signature/Descriptor parameters from JavaWriter ctors -> now taken from JavaCompiledMethod

kindlich committed:
Remove nameVariables parameter from ctors

kindlich committed:
Remove unused variable

kindlich committed:
Use proper CodePosition for Enum builtin methods (values/valueOf)

stanhebben commented on CodeModel/src/main/java/org/openzen/zenscript/codemodel/identifiers/DefinitionSymbol.java:
```diff
@@ -17,4 +17,5 @@ public interface DefinitionSymbol {
 
 	Optional<TypeSymbol> asType();
 
+    boolean isEnum();
```
Not really necessary here - you can do asType() instead and have it default to false. More logical to use the one in TypeSymbol instead.

stanhebben committed:
- Removed asEnum() in DefinitionSymbol
- Fixed resolving the enum type in Enum values
- Fixed locally defined types not being resolved

stanhebben committed:
Removed blocking todo
